### DISCUSSION TITLE
Is it possible to send an interaction without having a content?

### DISCUSSION THREAD

**Discussion by remeoj** [discordjs/discord.js#6119] [Q&A] [answered]
So I'm trying to send an embed as a result of an interaction. The issue is that I can't send a post callback to the interaction unless theres a message. Is there a way I can attach the embed as part of the message body or return an interaction resolve so that on discord I don't get the error "Interaction failed"? Any help is appreciated thanks!

**Comment by monbrey** [comment, marked as answer]
Sounds like you might be sending the embed incorrectly, as it should be an acceptable response.

```js
interaction.reply({ embeds: [yourEmbed] })
```

**Reply by remeoj** [reply]
okay thanks! Do you know where I could find this in the documentation? The issue is that I do Client.ws.on("INTERACTION_CREATE", func) and I'm not sure how to defer the interaction or send a resolve promise. When I tried doing the above, it says that interaction.reply() is not a function

**Reply by advaith1** [reply]
you should use discord.js v13 and the d.js interaction event (`client.on('interactionCreate', func)`)
docs: https://discord.js.org/#/docs/main/master/class/Client?scrollTo=e-interactionCreate
guide: https://v13-prep.discordjs.guide/interactions/replying-to-slash-commands.html
you can also ask in #djs-slash-commands in https://discord.gg/djs

**Reply by almostSouji** [reply]
Above should be the marked answer, but GitHub only allows toplevel responses to be marked so here we go.
You should be using the library native approach to handling interactions

**Reply by remeoj** [reply]
Hi Everyone! Thank you all for the help! I'll try the above and see if that works! Much appreciated!

**Reply by remeoj** [reply]
> you should use discord.js v13 and the d.js interaction event (`client.on('interactionCreate', func)`)
> docs: https://discord.js.org/#/docs/main/master/class/Client?scrollTo=e-interactionCreate
> guide: https://v13-prep.discordjs.guide/interactions/replying-to-slash-commands.html
> you can also ask in #djs-slash-commands in https://discord.gg/djs

Im trying to follow the guide https://v13-prep.discordjs.guide/interactions/registering-slash-commands.html but I'm confused why there are question marks in the code after application. I'm not sure if I should remove the question marks or if theres something I'm missing on instantiation

**Reply by almostSouji** [reply]
The question marks are valid in javascript as of node 14 and called [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). We will be using language features up to the current minimum required node version, which for v13 of discord.js is currently 14.0.0 or newer.
